pandas/tests/io/test_html.py

```diff
@@ -931,3 +936,32 @@ def test_same_ordering():
     dfs_lxml = read_html(filename, index_col=0, flavor=['lxml'])
     dfs_bs4 = read_html(filename, index_col=0, flavor=['bs4'])
     assert_framelist_equal(dfs_lxml, dfs_bs4)
+
+
+class ErrorThread(threading.Thread):
+    def run(self):
+        try:
+            super(ErrorThread, self).run()
+        except Exception as e:
+            self.err = e
+        else:
+            self.err = None
+
+
+@pytest.mark.slow
+def test_importcheck_thread_safety():
```

Comment:
I've just tested it with ``python3 `which pytest` test_html.py::test_importcheck_thread_safety`` and  ``python2 `which pytest` test_html.py::test_importcheck_thread_safety`` for both cases (with and without fix). The test seems to work for both versions on my system.